de.variantsync.core/src/test/java/de/variantsync/core/ASTTest.java

```diff
@@ -0,0 +1,53 @@
+package de.variantsync.core;
+
+import static org.junit.Assert.assertEquals;
+import java.util.Arrays;
+
+import de.variantsync.core.ast.AST;
+import de.variantsync.core.ast.LineGrammar;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ASTTest {
+
+	AST<LineGrammar, String> ast;
+	AST<LineGrammar, String> mainJava;
+
+	@Before
+	public void setup() {
+		ast = new AST<>(LineGrammar.Directory, "src");
+		AST<LineGrammar, String> mainDir = new AST<>(LineGrammar.Directory, "main");
+		AST<LineGrammar, String> testDir = new AST<>(LineGrammar.Directory, "test");
+		mainJava = new AST<>(LineGrammar.TextFile, "Main.java");
+		AST<LineGrammar, String> emptyJava = new AST<>(LineGrammar.TextFile, "Empty.java");
+		AST<LineGrammar, String> emptyTetJava = new AST<>(LineGrammar.TextFile, "EmptyTest.java");
+		ast.addChild(testDir);
+		testDir.addChild(emptyTetJava);
+		ast.addChild(mainDir);
+		mainDir.addChild(mainJava);
+		mainDir.addChild(emptyJava);
+
+		mainJava.addChildren(
+				Arrays.asList(new AST<>(LineGrammar.Line, "public class Main {"), new AST<>(LineGrammar.Line, "    public static void main(String[] args)"),
+						new AST<>(LineGrammar.Line, "        System.out.println(\"Hello World\");"), new AST<>(LineGrammar.Line, "    }"),
+						new AST<>(LineGrammar.Line, "}")));
+	}
+
+	@Test
+	public void sizeTest() {
+		assertEquals(11, ast.size());
```

Comment:
This requires the tests to run in a specifc order, right? This is verified by the java annotations? I would prefer if each test is side-effect free but that might be unnecessary or unperfomant? Convince me of the opposite! :)

Reply:
Ok convinced.